cmd/config/internal/commands/cmdset.go

```diff
@@ -116,7 +130,7 @@ func (r *SetRunner) runE(c *cobra.Command, args []string) error {
 		fmt.Fprintf(c.OutOrStdout(), "set %d fields\n", count)
 		return handleError(c, err)
 	}
-	if len(args) == 3 {
+	if len(args) == 2 && r.Set.Value != "" || len(args) == 3 {
```

Comment:
this can just be `len(args) > 2 || IsFlagSet("value")` -- cobra checks that it is at least 2 for us

Reply:
Done, test added.